### ISSUE TITLE
over-eager method deletion?

### ISSUE THREAD

**Issue by vtjnash** [timholy/Revise.jl#137]
I had some code like this in a submodule, and it seems like Revise.jl deleted these method when I loaded this package (and touched the file). Sorry I can't try to check it further as an MRE right now.
```julia
let
    indent = "  ︰  "^10
    indent_z = collect(eachindex(indent))
    global function print_indent(io::IO, depth::Int)
        depth < 1 && return
        while depth > length(indent_z)
            print(io, indent)
            depth -= length(indent_z)
        end
        print(io, SubString(indent, 1, indent_z[depth]))
    end
end

function process(data::Vector{<:Unsigned}=UInt[], lidict::Set=Base.Set(data))
end
```

**Comment by timholy** [comment]
Thanks for the example. Sadly doesn't  trigger for me:
```julia
julia> includet("nash.jl")

julia> process
process (generic function with 3 methods)

julia> methods(process)
# 3 methods for generic function "process":
[1] process() in Main at /tmp/nash.jl:15
[2] process(data::Array{#s24,1} where #s24<:Unsigned) in Main at /tmp/nash.jl:15
[3] process(data::Array{#s25,1} where #s25<:Unsigned, lidict::Set) in Main at /tmp/nash.jl:15

julia> print_indent
print_indent (generic function with 1 method)

shell> touch nash.jl

julia> 

julia> print_indent
print_indent (generic function with 1 method)

julia> methods(process)
# 3 methods for generic function "process":
[1] process() in Main at /tmp/nash.jl:15
[2] process(data::Array{#s24,1} where #s24<:Unsigned) in Main at /tmp/nash.jl:15
[3] process(data::Array{#s25,1} where #s25<:Unsigned, lidict::Set) in Main at /tmp/nash.jl:15

julia> print_indent(stdout, 5)
  ︰  
```
True also if I place the above inside
```julia
module Mod1

module Mod2
<body>

end

end
```
Happy to go over these in person together when you have time.

**Comment by vtjnash** [comment]
Could it be related that these were in stdlib?

**Comment by timholy** [comment]
Maybe. #163 might help debug stuff.

**Comment by timholy** [comment]
I'm calling this fixed by #180. If you encounter future troubles you can always [have logging on by default](https://timholy.github.io/Revise.jl/latest/debugging.html#Logging-by-default-1) and consequently not worry so much about whether you can find a MWE.
